File: 2284-smallest-value-of-the-rearranged-number/smallest-value-of-the-rearranged-number.py
class Solution:
    def smallestNumber(self, num: int) -> int:
        if num==0: return 0
        if num<0:
            s="".join(sorted(str(-num),reverse=True))
            return -int(s)
        s=sorted(list(str(num)))
        i=0
        while s[i]=='0':
            i+=1
        s[0],s[i]=s[i],s[0]
        return int("".join(s))

        

File: 2284-smallest-value-of-the-rearranged-number/smallest-value-of-the-rearranged-number.cpp
class Solution {
public:
    long long smallestNumber(long long num) {
        if(num==0) return 0;
        string s=to_string(num);
        if(num<0){
            sort(s.rbegin(),s.rend());
            return stoll(s)*-1;
        }
        sort(s.begin(),s.end());
        int i=0;
        while(s[i]=='0') i++;
        swap(s[0],s[i]);
        return stoll(s);
    }
};